File: java-completed/app/src/main/java/com/vonage/tutorial/voice/BaseApplication.java
package com.vonage.tutorial.voice;

import android.app.Application;
import com.nexmo.client.BuildConfig;
import com.nexmo.client.NexmoClient;
import timber.log.Timber;

public final class BaseApplication extends Application {

    public void onCreate() {
        super.onCreate();

        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }

        this.initializeNexmoClient();
    }

    private final void initializeNexmoClient() {
        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        new NexmoClient.Builder().build(this);
    }
}

File: kotlin-completed/app/src/main/java/com/vonage/tutorial/voice/BaseApplication.kt
package com.vonage.tutorial.voice

import android.app.Application
import com.nexmo.client.BuildConfig
import com.nexmo.client.NexmoClient
import timber.log.Timber

class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }

        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        NexmoClient.Builder()
            .useFirstIceCandidate(true)
            .build(this)
    }
}